### PR TITLE
Add HDF5 input and output to reduction_to_band miniapp

### PR DESCRIPTION
**Pull request by RMeli**
Add HDF5 input and output files to `reduction_to_band` miniapp.

### CODE REVIEW

**Comment by RMeli** [comment]
Waiting for the discussion on #973 to happen.

File: miniapp/miniapp_reduction_to_band.cpp [hunk 1]
```diff
@@ -81,19 +105,25 @@ struct reductionToBandMiniapp {
...
 
-    // Allocate memory for the matrix
-    const GlobalElementSize matrix_size(opts.m, opts.m);
-    const TileElementSize block_size(opts.mb, opts.mb);
-
-    ConstMatrixType matrix_ref = [matrix_size, block_size, comm_grid]() {
+    ConstMatrixType matrix_ref = [comm_grid, &opts]() {
+#ifdef DLAF_WITH_HDF5
+      if (opts.input_file) {
+        Matrix<T, Device::CPU> input = opts.input_file->read<T>("/a", {opts.mb, opts.mb});
```
Comment (RMeli) [inline miniapp/miniapp_reduction_to_band.cpp:111]:
```suggestion
        Matrix<T, Device::CPU> input = opts.input_file->read<T>("/input", {opts.mb, opts.mb});
```
?

File: miniapp/miniapp_reduction_to_band.cpp [hunk 2]
```diff
@@ -136,6 +166,14 @@ struct reductionToBandMiniapp {
         gigaflops = dlaf::total_ops<T>(add_mul, add_mul) / elapsed_time / 1e9;
       }
 
+#ifdef DLAF_WITH_HDF5
+      if (run_index == opts.nruns - 1) {
+        if (opts.output_file) {
+          opts.output_file->write<T>(matrix_host, "/band");
```
Comment (RMeli) [inline miniapp/miniapp_reduction_to_band.cpp:172]: ?

File: miniapp/miniapp_reduction_to_band.cpp (hunk 1 above)
Comment (RMeli) [reply miniapp/miniapp_reduction_to_band.cpp]: Added `input-dataset` parameter, with `/input` as default.

**Comment by RMeli** [comment]
cscs-ci run

**Review by rasolca — APPROVED** [review]

**Review by msimberg — APPROVED** [review]

**Pull request merged**